semantic-core/src/Data/Scope.hs

```diff
@@ -105,3 +108,13 @@ instantiate f = instantiateEither (either f pure)
 
 instantiateEither :: Monad f => (Either a b -> f c) -> Scope a f b -> f c
 instantiateEither f = unScope >=> incr (f . Left) (>>= f . Right)
+
+
+un :: (Int -> t -> Maybe (a, t)) -> t -> (Stack a, t)
+un from = unEither (matchMaybe . from)
```

Comment:
These helpers enable us to decompose right-nested chains of Core constructors, typically syntax binders, using a higher-order function to select matching terms. I’m using something similar in `path`.

Reply:
The name is a little too concise for my tastes. Maybe `prune` or something? Also stick the above as a Haddock comment please 🤗

Reply:
Renamed to `unprefix` & documented 👍